src/Integration.UnitTests/MefServices/ActiveSolutionBoundTrackerTests.cs

```diff
@@ -568,5 +574,44 @@ private void VerifyAndResetBoundSolutionUIContextMock(bool isActive)
             vsMonitorMock.Verify(x => x.SetCmdUIContext(It.IsAny<uint>(), It.IsAny<int>()), Times.Once);
             vsMonitorMock.Invocations.Clear();
         }
+
+        private class EventCounter
```

Comment:
Simple helper - counts each of the raised events, and records the order in which they are raised.